### DISCUSSION TITLE
Way to pause animations.

### DISCUSSION THREAD

**Discussion by slavko-lukic** [software-mansion/react-native-reanimated#5342] [Ideas]
In some cases, I think it would be useful to pause animations, similarly like you can cancel now.
I imagine it would work something like this:

```javascript
const offset = useSharedValue(0)

const startAnimation = () => {
    offset.value = withTiming(1, { duration: 10000 });
}

const pause = () => {
    pauseAnimation(offset);
}

const resume = () => {
    resumeAnimation(offset);
}
```

**Comment by wcandillon** [comment]
We have built it in redash: https://wcandillon.gitbook.io/redash/animations#withpause

**Reply by slavko-lukic** [reply]
Thanks, I will take a look at redash, seems like a good option.

**Comment by tunahanbayindir** [comment]
You can call `offset.value = offset.value`

**Reply by tomekzaw** [reply]
This will stop the animation without the ability to resume it.

**Reply by tunahanbayindir** [reply]
Just assign the destination again, and it'll continue from there. (idk if it has any underlying side effect tho)

**Reply by MatiPl01** [reply]
@tunahanbayindir Just to make sure, you are suggesting doing something like this:

```ts
offset.value = withTiming(...); // to start the animation for the first time

offset.value = offset.value; // to stop the animation

offset.value = withTiming(...); // to resume the animation
```

If that is what you suggest, then this solution is not correct. The second assignment of the `withTiming` function call will start the new animation from the last `offset` value but won't preserve the elapsed time and progress of the previous animation. Even if you calculate the remaining duration of the paused animation manually and adjust the duration in the second `withTiming` call, you won't get the same effect because of the easing (unless you set easing to `Easing.linear`).

If you meant something else than this, please clarify your suggestion.
